### ISSUE TITLE
Design proposal

### ISSUE THREAD

**Issue by cosenal** [Apophenia/news-firefox#1]
The add-on should have the following user workflow:
- an rss icon in the location bar should appear for sites that expose an rss feed (like the one that chrome and firefox used to have back in the old days of Google Reader);
  For inspiration, see:
  [1] https://github.com/justinkelly/chrome-rss
  [2] https://github.com/JasnaPaka/rss-icon-in-awesombar
- once the user clicks on the icon, a confirmation modal should pop up. The user would select the folder from the modal, click on »Add« and the feed would be added to the News app.
- the oC server url and oC user account information should be stored in the Preferences settings of the add-on.

This is just my idea, of course @jancborchardt can do better ;-)

cc @Raydiation

**Comment by jancborchardt** [comment]
Some improvement suggestions for the flow:
- Instead of the RSS icon being shown, it should be an ownCloud icon (or maybe a combination of ownCloud and the RSS quarter-circles). Because it’s specifically ownCloud.
- When clicking the icon, the feed should automatically be added to the root folder. Feedback and choice of folders should be after that in a small popover which will vanish after a few seconds (say 5) if not interacted with. (For comparison, check out how Pocket does that – assigning tags is completely optional, and the overlay vanishes after a few seconds.)
- If the account is not connected yet (first use), the popover should contain an input for ownCloud server address, username and password to connect to it.

What do you think? @cosenal @Apophenia

**Comment by cosenal** [comment]
See, I knew you would do better than me :p 

I hereby declare impossible to combine the RSS icon with the ownCloud icon! Whoever has tried has failed :( Maybe just the logo of the News app instead?

**Comment by jancborchardt** [comment]
Yeah, the logo is a detail though. Maybe the icon of the news app works, although that’s pretty generic. Anyway, detail for later.

**Comment by cosenal** [comment]
Probably it's nice to show a preview of the feed too, if that's easy to do with a bit of js.

**Comment by Apophenia** [comment]
Thanks for all of these suggestions. I'm going to go ahead and start working on storing credentials in preferences today, but I'll also play around with Pocket et cetera to get a feel for the kind of functionality we're looking for.
